grpc/src/test/java/com/linecorp/armeria/client/grpc/GrpcServiceImplErrorFromClientViewTest.java

```diff
@@ -0,0 +1,325 @@
+        @Override
+        public void unaryCall(SimpleRequest request, StreamObserver<SimpleResponse> responseObserver) {
+            responseObserver.onError(new StatusRuntimeException(Status.ABORTED, corruptedMetadata));
+        }
+
+        // blocking client
+        // Error happens when serializing metadata.
+        @Override
+        public void streamingOutputCall(StreamingOutputCallRequest request,
+                                        StreamObserver<StreamingOutputCallResponse> responseObserver) {
+            responseObserver.onNext(StreamingOutputCallResponse.newBuilder().setPayload(
+                    Payload.newBuilder().setBody(ByteString.copyFromUtf8("pong1")).build()).build());
+            if (request.equals(STREAMING_OUTPUT_CALL_REQUEST_MESSAGE)) {
+                responseObserver.onError(
+                        new StatusRuntimeException(Status.FAILED_PRECONDITION, corruptedMetadata));
+                return; // onError must be the last call.
+            }
+            // The following lines are not reached.
+            responseObserver.onNext(StreamingOutputCallResponse.newBuilder().setPayload(
+                    Payload.newBuilder().setBody(ByteString.copyFromUtf8("pong2")).build()).build());
+            responseObserver.onCompleted();
+        }
+
+        // Error happens when serializing metadata.
+        @Override
+        public StreamObserver<StreamingOutputCallRequest> fullDuplexCall(
+                StreamObserver<StreamingOutputCallResponse> responseObserver) {
+            return new StreamObserver<StreamingOutputCallRequest>() {
+                int cnt;
+
+                @Override
+                public void onNext(StreamingOutputCallRequest value) {
+                    // Return error on the 3rd message.
+                    if (++cnt == 3) {
+                        responseObserver.onError(
+                                new StatusRuntimeException(Status.FAILED_PRECONDITION, corruptedMetadata));
+                    } else {
+                        responseObserver.onNext(
+                                StreamingOutputCallResponse
+                                        .newBuilder()
+                                        .setPayload(
+                                                Payload.newBuilder()
+                                                       .setBody(ByteString.copyFromUtf8("pong" + cnt)).build())
+                                        .build());
+                    }
+                }
+
+                @Override
+                public void onError(Throwable t) {
+                    // no-op
+                }
+
+                @Override
+                public void onCompleted() {
+                    // no-op
+                }
+            };
+        }
+    }
+
+    // Use this method to ensure to escape assertion in the constructor of Metadata.
+    private static boolean isSameSizeAndNameAndValues(Metadata a, Metadata b) {
```

Comment:
I think just using `toString` to compare the metadata is sufficient 😄

Reply:
Yeah, if I could, I'll do as you said.
I suspect `toString`  might use invalid `size` field to serialize the field `nameAndValues`, so `IndexOutOfBoundsException` would be thrown. (I didn't check it strictly. I'll check the way of `toString` anyway.)

As far as I've seen, `assertThat#isEqualTo`  fails for `corruptedMetadata`.

Reply:
I've checked, but it fails with `IndexOutOfBoundsException` as I've said unfortunately..